tools/dynamic_framework_slicer/dynamic_framework_slicer.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash 
+set -eu
+
+BINARIES+=()
+while [[ $# -gt 0 ]]; do
+  arg="$1"
+  case $arg in
+      --in)
+      readonly IN="$2"
+      shift
+      ;;
+      --out)
+      readonly OUT="$2"
+      shift
+      ;;
+      --realpath)
+      readonly REALPATH="$2"
+      shift
+      ;;
+      *)
+      BINARIES+=("$1")
+      ;;
+  esac
+  shift
+done
+
+# Strip out any unnecessary slices from embedded dynamic frameworks to save space
```

Comment:
Sanity check args?  Error if missing IN/BINARIES/etc?

i.e. - if `IN` were a directory or a nonexistent file, won't this be a noop, is that good?

Reply:
I don't know enough to say whether it should be an error or a no-op. I will add some argument checking to ensure the types are what we expect, and that we have arguments, but I'm not entirely sure if there are valid builds where the script will be called with the wrong arguments.

Reply:
In general, be a little defense just so if someone updates the bzl without updating the script to match, things fail vs. accidentally do nothing.